examples/seismic/viscoacoustic/operators.py

```diff
@@ -63,7 +62,85 @@ def blanch_symes(model, geometry, v, p, **kwargs):
     return [u_v, u_r, u_p]
 
 
-def ren(model, geometry, v, p, **kwargs):
+def sls_2nd_stencil(model, geometry, v, p, **kwargs):
+    """
+    Stencil created from Bai (2014) viscoacoustic wave equation.
+
+    https://library.seg.org/doi/10.1190/geo2013-0030.1
+
+    Parameters
+    ----------
+    v : VectorTimeFunction
+        Particle velocity.
+    p : TimeFunction
+        Pressure field.
+    """
+    forward = kwargs.get('forward')
+    space_order = p.space_order
+    s = model.grid.stepping_dim.spacing
+    b = model.b
+    vp = model.vp
+    damp = model.damp
+    qp = model.qp
+    f0 = geometry._f0
+
+    # The stress relaxation parameter
+    t_s = (sp.sqrt(1.+1./qp**2)-1./qp)/f0
+
+    # The strain relaxation parameter
+    t_ep = 1./(f0**2*t_s)
+
+    # The relaxation time
+    tt = (t_ep/t_s)-1.
+
+    # Density
+    rho = 1. / b
+
+    r = TimeFunction(name="r", grid=model.grid, time_order=2, space_order=space_order,
```

Comment:
Shouldn't this be part of `model`? If it was the appropriate stencil and case could be constricted depending on what fields and `kwargs` are passed right?

Reply:
The variable `b` is already part of the model. The` rho` is computed from `b`.